### ISSUE TITLE
[New Rule] Proxies and interceptors MUST never be explicitly requested in constructors

### ISSUE THREAD

**Issue by lenaorobei** [magento/magento-coding-standard#18]
### Rule
Proxies and interceptors MUST never be explicitly requested in constructors.

Source - [Magento Technical Guidelines](https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html)

### Description
Proxies are used for lazy-loading. Details on [DevDocs](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/proxies.html).

Proxies should be configured via `di.xml` file.
```
<type name="FastLoading">
    <arguments>
        <argument name="slowLoading" xsi:type="object">SlowLoading\Proxy</argument>
    </arguments>
</type>
```

Plugins (interceptors) modify the behavior of public class functions by intercepting a function call and running code before, after, or around that function call. Details on [DevDocs](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/plugins.html).

Plugins should be declared via `di.xml` file.

```
<config>
    <type name="{ObservedType}">
      <plugin name="{pluginName}" type="{PluginClassName}" sortOrder="1" disabled="false" />
    </type>
</config>
```
### Implementation
Subscribe to the `T_CLASS` token and analyse `__constructor` arguments.
```xml
<severity>8</severity>
<type>warning</type>
```

**Comment by maderlock** [comment]
After discussion in pull-request, the scope of this has changed slightly:
1. This will output errors instead of warnings
2. Rather than detect plugins, the interceptors themselves will be detected
